### PR TITLE
Fix compiler warnings, part 3

### PR DESCRIPTION
**Pull request by nilason**
Fixes compiler warnings:

- -Wnon-literal-null-conversion
- -Wmacro-redefined
- -Wpointer-bool-conversion

Third part addressing #1247.

Please see comments in code for fixes of `-Wpointer-bool-conversion`warnings.

Modules / code parts directly affected:

- db/db.select
- lib/db/dbmi_base
- vector/v.category
- display/d.mon
- imagery/i.landsat.toar
- include/iostream

### CODE REVIEW

File: display/d.mon/list.c [hunk 1]
```diff
@@ -52,7 +52,8 @@ void list_mon(char ***list, int *n)
     while ((dp = readdir(dirp)) != NULL) {
 	int ret;
 
-        if (!dp->d_name || dp->d_name[0] == '.')
+        /* 'dp->d_name' will always be true... */
+        if (!*dp->d_name || dp->d_name[0] == '.')
```
Comment (metzm) [inline display/d.mon/list.c:56]:
```suggestion
        if (!dp->d_name[0] || dp->d_name[0] == '.')
```
Just for clarification in the code that d_name is an array, not a pointer, see `man 3 readdir`.

File: imagery/i.landsat.toar/landsat_set.c [hunk 2]
```diff
@@ -45,7 +45,10 @@ void sensor_TM(lsat_data * lsat)
     double wmax[] = { 0.52, 0.60, 0.69, 0.90, 1.75, 12.50, 2.35 };
     /* 30, 30, 30, 30, 30, 120 original, 60 resamples before Feb 25, 2010 and 30 after, 30 */
 
-    if (!lsat->sensor)
+    /* address of array 'lsat->sensor' will always evaluate to 'true' */
+    /* In this case the following condition is always false and */
+    /* strcpy() is never reached */
+    if (!*lsat->sensor)
```
Comment (metzm) [inline imagery/i.landsat.toar/landsat_set.c:51]: Looking at the other `sensor_*()`fns, it should be
```
    strcpy(lsat->sensor, "TM");
```
the `if (!lsat->sensor)` before must be removed.

File: imagery/i.landsat.toar/main.c [hunk 3]
```diff
@@ -319,7 +319,8 @@ int main(int argc, char *argv[])
 	G_debug(1, "lsat.number = %d, lsat.sensor = [%s]",
 		lsat.number, lsat.sensor);
 
-	if (!lsat.sensor || lsat.number > 8 || lsat.number < 1)
+	/* address of array 'lsat.sensor' will always evaluate to 'true' */
+        if (!*lsat.sensor || lsat.number > 8 || lsat.number < 1)
```
Comment (metzm) [inline imagery/i.landsat.toar/main.c:323]:
```suggestion
        if (!lsat.sensor[0] || lsat.number > 8 || lsat.number < 1)
```
lsat.sensor is an array `char sensor[10];`

File: lib/db/dbmi_base/login.c [hunk 4]
```diff
@@ -297,10 +297,10 @@ static int get_login(const char *driver, const char *database, const char **user
 
     G_debug(3, "db_get_login(): drv=[%s] db=[%s]", driver, database);
 
-    user[0] = '\0';
-    password[0] = '\0';
-    host[0] = '\0';
-    port[0] = '\0';
+    user[0] = NULL;
```
Comment (metzm) [inline lib/db/dbmi_base/login.c:300]:
```suggestion
    *user = NULL;
    *password = NULL;
    *host = NULL;
    *port = NULL;
```
These are pointers to pointers, see code below.

File: display/d.mon/list.c (hunk 1 above)
Comment (nilason) [reply display/d.mon/list.c]: Yes, `*db->d_name` will only work for instantiated strings. My bad. Updated as suggested.

File: imagery/i.landsat.toar/landsat_set.c (hunk 2 above)
Comment (nilason) [reply imagery/i.landsat.toar/landsat_set.c]: Updated as suggested.

File: imagery/i.landsat.toar/main.c (hunk 3 above)
Comment (nilason) [reply imagery/i.landsat.toar/main.c]: Updated as suggested.

File: lib/db/dbmi_base/login.c (hunk 4 above)
Comment (nilason) [reply lib/db/dbmi_base/login.c]: Updated as suggested.

**Comment by nilason** [comment]
Updated the suggested changes.

Ready for review. For the time being I left the comments in code, but I will remove them before merge.

**Review by metzm — APPROVED** [review]
From my side, ready to merge, thanks a lot!

**Comment by nilason** [comment]
Cleaned up comments in code and reworded the commit msgs.
Ready for merge.

**Review by metzm — APPROVED** [review]
Ready for merge, thanks!

**Comment by nilason** [comment]
Cherry picked and pushed manually.

Thanks @metzm !

**Pull request closed without merging**